### PR TITLE
feat: use RxJS pipeable operators

### PR DESCRIPTION
**Pull request by DavidRouyer**

### CODE REVIEW

File: src/ngx-uploader/classes/ngx-uploader.class.ts [hunk 1]
```diff
@@ -26,13 +26,15 @@ export class NgUploaderService {
...
     this.queue = [];
-    this.serviceEvents = new EventEmitter<any>();
+    this.serviceEvents = new EventEmitter<UploadOutput>();
     this.uploadScheduler = new Subject();
     this.subs = [];
     this.contentTypes = contentTypes;
 
     this.uploadScheduler
-      .mergeMap(upload => this.startUpload(upload), concurrency)
+      .pipe(
```
Comment (jkuri) [inline src/ngx-uploader/classes/ngx-uploader.class.ts]: hi @DavidRouyer. sorry for super late response. can you explain what is this different from the current implementation? thanks.

Comment (DavidRouyer) [reply src/ngx-uploader/classes/ngx-uploader.class.ts]: Check the docs [here](https://github.com/ReactiveX/rxjs/blob/master/doc/pipeable-operators.md). Pipeable operators were introduced in RxJS 5.5 and you use them like this.

**Comment by jkuri** [comment]
rebase please.

**Comment by sumpton** [comment]
should this
> import { mergeMap } from 'rxjs/operators/mergeMap';

be

> import { mergeMap } from 'rxjs/operators';

**Comment by DavidRouyer** [comment]
@sumpton If I am correct, `import { mergeMap } from 'rxjs/operators/mergeMap';` will only include the mergeMap operator when tree shaked while `import { mergeMap } from 'rxjs/operators';` will include all the operators in `rxjs/operators` for now. You can take a look at the [material2](https://github.com/angular/material2) repo, they do it that way.

@jkuri Is it good for you?

**Comment by jkuri** [comment]
great. thank you.

**Comment by sumpton** [comment]
@DavidRouyer Thanks. Interesting. I see on the rxjs site they nest the function down a level, where their examples [here](https://github.com/ReactiveX/rxjs/blob/master/doc/pipeable-operators.md) are the destructuring I expected. I appreciate learning this.

**Pull request merged**